mongo/integration/server_selection_prose_test.go

```diff
@@ -107,6 +115,11 @@ func TestServerSelectionProse(t *testing.T) {
 			SetHosts(hosts[:2]).
 			SetPoolMonitor(tpm.PoolMonitor))
 
+		// Sleep for 100ms to allow all server state discovery to complete. We need both servers to
+		// be selectable when we start running the test or the distribution of selected servers will
+		// be skewed. Unfortunately there's not currently another signal we can block on.
+		time.Sleep(100 * time.Millisecond)
```

Comment:
Suggested offline, consider waiting for two `ServerHeartbeatSucceededEvent` events rather than sleep.

Reply:
Actually used `TopologyDescriptionChangedEvent`s to confirm that the topology has exactly 2 `Mongos` type instances before continuing to run the test.

Reply:
Updated.